pandera/engines/ibis_engine.py

```diff
@@ -0,0 +1,107 @@
+"""Ibis engine and data types."""
+
+import dataclasses
+import inspect
+import warnings
+from typing import Any, Iterable, Union
+
+import ibis
+import ibis.expr.datatypes as dt
+import numpy as np
+
+from pandera import dtypes
+from pandera.dtypes import immutable
+from pandera.engines import engine, numpy_engine
+
+
+@immutable(init=True)
+class DataType(dtypes.DataType):
+    """Base `DataType` for boxing Ibis data types."""
+
+    type: Any = dataclasses.field(repr=False, init=False)
+    """Native Ibis dtype boxed by the data type."""
+
+    def __init__(self, dtype: Any):
```

Comment:
Don't think I've written any code that hits this yet.

Reply:
https://github.com/unionai-oss/pandera/pull/1451/files#diff-6bd99d89ccace74b1b743c0d34ae391325b1b194a0e11c7dcbb478513d353c7aR96-R99 should translate into this datatype via the `Engine` class